### PR TITLE
Support navigating to metadata in progression results

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/7151

### CODE REVIEW

File: src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs [hunk 1]
```diff
@@ -32,74 +30,72 @@ internal sealed class GraphNavigatorExtension(
...
-                return;
-            }
+        _threadingContext.JoinableTaskFactory.Run(() => NavigateToAsync(graphNode, CancellationToken.None));
+    }
 
-            var projectId = graphNode.GetValue<ProjectId>(RoslynGraphProperties.ContextProjectId);
-            var symbolId = graphNode.GetValue<SymbolKey?>(RoslynGraphProperties.SymbolId);
+    private async Task NavigateToAsync(GraphNode graphNode, CancellationToken cancellationToken)
+    {
+        var projectId = graphNode.GetValue<ProjectId>(RoslynGraphProperties.ContextProjectId);
```
Comment (dibarbet) [inline src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs:41]: Are there any UI thread requirements for this?

File: src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs [hunk 2]
```diff
@@ -32,74 +30,72 @@ internal sealed class GraphNavigatorExtension(
...
-                        StringComparison.OrdinalIgnoreCase));
+        var solution = _workspace.CurrentSolution;
+        var project = solution.GetProject(projectId);
+        if (project is null)
+            return;
 
-                if (document == null)
+        // Go through the mainline symbol id path if we have it.  That way we notify third parties, and we can navigate
+        // to metadata.
+        var symbolId = graphNode.GetValue<SymbolKey?>(RoslynGraphProperties.SymbolId);
```
Comment (dibarbet) [inline src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs:53]: Do we ever get called for other languages (TS/F#) here?

**Review by dibarbet — APPROVED** [review]

File: src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs (hunk 2 above)
Comment (dibarbet) [reply src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs:53]: looks the same as before, so assumingthis is fine

File: src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs:41]: nope!

File: src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs:53]: nope!

**Pull request merged**